src/main/java/com/epam/javacourses2016/task13/Edge.java

```diff
@@ -0,0 +1,48 @@
+package com.epam.javacourses2016.task13;
+
+/**
+ * Created by Ivan on 24/11/2016.
+ */
+public class Edge {
+
+    private int first;
+    private int second;
+
+    public Edge(int first, int second) {
+        this.first = first;
+        this.second = second;
+    }
+
+    public int getFirst() {
+        return first;
+    }
+
+    public int getSecond() {
+        return second;
+    }
+
+    @Override
+    public int hashCode() {
+        final int PRIME = 31;
+        int result = 1;
+        result = PRIME * result + first;
+        result = PRIME * result + second;
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj)
+            return true;
+        if (obj == null)
+            return false;
+        if (getClass() != obj.getClass())
+            return false;
```

Comment:
1. During the convention, comparing with null is always false, that part is right. But in case we compare two different objects we should throw some RuntimeException

3. No, actually me review is just recomendation